### ISSUE TITLE
My own messages are not appearing as verified to me in #megolm

### ISSUE THREAD

**Issue by ara4n** [vector-im/vector-web#2232]

**Comment by ara4n** [comment]
Looks like this is because we're not setting keysProved on events that we send.  A workaround seems to be:

```
diff --git a/lib/crypto/index.js b/lib/crypto/index.js
index 28b65ca..618ced5 100644
--- a/lib/crypto/index.js
+++ b/lib/crypto/index.js
@@ -797,6 +797,10 @@ Crypto.prototype.encryptEventIfNeeded = function(event, room) {
         return null;
     }
 
+    event.getClearEvent().keysProved = {
+        'curve25519' : this._olmDevice.deviceCurve25519Key
+    };
+
     return alg.encryptMessage(
         room, event.getType(), event.getContent()
     ).then(function(encryptedContent) {
```

...however, surely we need to be setting *actual* keysProved and keysClaimed on our outbound events, or this isn't going to work at all?

**Comment by NegativeMjark** [comment]
Fixed by matrix-org/matrix-js-sdk#210.
